src/ssa/address_canonizer.cpp

```diff
@@ -40,22 +40,25 @@ exprt address_canonizer(
     else if(object.id()==ID_member)
     {
       // get offset
-      exprt offset=member_offset_expr(to_member_expr(object), ns);
-
-      // &x.m  ---> (&x)+offset
-
-      address_of_exprt address_of_expr(to_member_expr(object).struct_op());
-      exprt rec_result=address_canonizer(address_of_expr, ns); // rec. call
-
-      pointer_typet byte_pointer(
-        unsigned_char_type(),
-        config.ansi_c.pointer_width);
-      typecast_exprt typecast_expr(rec_result, byte_pointer);
-      plus_exprt sum(typecast_expr, offset);
-      if(sum.type()!=address.type())
-        sum.make_typecast(address.type());
-
-      return sum;
+      if(auto maybe_offset=member_offset_expr(to_member_expr(object), ns))
+      {
+        // &x.m  ---> (&x)+offset
+
+        address_of_exprt address_of_expr(to_member_expr(object).struct_op());
+        exprt rec_result=address_canonizer(address_of_expr, ns); // rec. call
+
+        pointer_typet byte_pointer(
+          unsigned_char_type(),
+          config.ansi_c.pointer_width);
+        typecast_exprt typecast_expr(rec_result, byte_pointer);
+        exprt result;
+        plus_exprt sum(typecast_expr, *maybe_offset);
+        result=sum;
+        if(sum.type()!=address.type())
+          result=typecast_exprt(sum, address.type());
+
+        return result;
+      }
```

Comment:
What does this branch return?

Reply:
As far as I can understand, it converts an expression where a member is accessed to an expression where offset is used instead of the member.